backend/app/main.py

```diff
@@ -13,25 +16,55 @@
 from pyjwt_key_fetcher import AsyncKeyFetcher
 from starlette.middleware.sessions import SessionMiddleware
 
-from .settings import conf
+key_fetcher = AsyncKeyFetcher()
+oauth = OAuth()
+
+
+async def register_oauth():
+    dataspace_configuration = await get_dataspace_configuration()
+    authentication_providers = dataspace_configuration["authentication_providers"]
+    authentication_providers_end_user = authentication_providers["end_user"]
+    authentication_providers_end_user_base_url = authentication_providers_end_user[
+        "base_url"
+    ]
+
+    oauth.register(
+        name="login_portal",
+        client_id=conf.OIDC_CLIENT_ID,
+        client_secret=conf.OIDC_CLIENT_SECRET,
+        server_metadata_url=(
+            authentication_providers_end_user_base_url
+            + "/.well-known/openid-configuration"
+        ),
+        client_kwargs={
+            "scope": conf.OIDC_SCOPES,
+            "timeout": Timeout(timeout=conf.OIDC_REQUEST_TIMEOUT),
+        },
+    )
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    await register_oauth()
```

Comment:
Something that I realized is that if you here in the lifespan would as the first thing fetch the dataspace configuration you could then in all other places just import and use `DATASPACE_CONFIGURATION` directly since it would be fetched at application startup and thus always available. Works like it is right now as well, so feel free to mark as resolved with no change.

Reply:
@joakimnordling that is assuming the dataspace configuration never changes, and it might

Reply:
True. Are you @lietu still fine then with the infinite storage of the `DATASPACE_CONFIGURATION` or would you want to see some sort of TTL cache being used for it as well? Without changing that it's merely providing the foundation for changing it in the future to use some TTL cache instead of infinite (runtime) cache.

Right now it merely replaces things that were configured in the settings of the app and thus not at all dynamically updated ever earlier, now it's still reloaded upon each start.

Reply:
I've been using an LRU cache so far but combining it with e.g. 30min TTL would be better https://github.com/ioxiocom/ioxio-tags/blob/1922f08f853e7a66cd832cf8c91620a3267d77f4/api/app/dataproduct.py#L11-L13

Reply:
I would generally prefer our examples to teach people the right way to build for dataspaces, which mean no infinite caches of files that may change.